### PR TITLE
Feat/cozy auth

### PR DESCRIPTION
**Pull request by Crash--**
Currently being tested

Several points:
- Use withClient() instead of this.context.client
- Use cozy-auth to handle auth 
- Add support for universal link 
- New way to handle the push to /onboarding. We store an `onboarded` boolean in the localStorage 
- Add a migration function to handle the old and new way to connect

### CODE REVIEW

File: src/drive/mobile/modules/authorization/DriveMobileRouter.jsx [hunk 1]
```diff
@@ -1,108 +1,93 @@
...
-  onboardingInformationsPropTypes,
-  onboardingPropTypes
-} from '../../../../authentication/OnboardingPropTypes'
+import appBooted from '../../../targets/mobile/'
 class DriveMobileRouter extends Component {
-  static contextTypes = {
-    client: Proptypes.object.isRequired
+  async componentDidMount() {
+    //Wait for the app to be booted to avoid race condition between cordova & JS
+    await appBooted
```
Comment (y-lohse) [inline src/drive/mobile/modules/authorization/DriveMobileRouter.jsx]: I'm not sure how that works, I thought awaiting in componentDidMount didn't prevent the initial rendering of the component..?

File: src/drive/mobile/modules/authorization/DriveMobileRouter.jsx [hunk 2]
```diff
@@ -1,108 +1,93 @@
...
-          appRoutes={appRoutes}
+          protocol={PROTOCOL}
+          appTitle={SOFTWARE_NAME}
+          universalLinkDomain={getUniversalLinkDomain()}
+          appSlug={appMetadata.slug}
           history={history}
-          onAuthenticated={this.afterAuthentication}
+          onAuthenticated={async () => {
+            return await this.afterAuthentication()
+          }}
```
[the remark this answers is not in the document]
Comment (Crash--) [reply src/drive/mobile/modules/authorization/DriveMobileRouter.jsx]: fixed, thanks

File: src/drive/mobile/modules/authorization/DriveMobileRouter.jsx (hunk 1 above)
Comment (Crash--) [reply src/drive/mobile/modules/authorization/DriveMobileRouter.jsx]: Oh you have totally right! changed to use a state instead

**Comment by y-lohse** [comment]
Last commit LGTM 👍

**Pull request merged**